templates/CMakeLists.txt.template

```diff
@@ -294,6 +305,12 @@
   add_custom_target(tools
     DEPENDS tools_c tools_cxx)
 
+  % for src in sorted(protobuf_gen_files):
+  protobuf_generate_grpc_cpp(
```

Comment:
just to check: code generation is only run if the resulting .cc files are actually needed by some of the targets being build? (we don't want all the codegen to just run upon every build).

Reply:
Correct. This is really no different from how it was before - the only difference is that `protobuf_generate_grpc_cpp()` isn't being called twice for the same file.